Makefile

```diff
@@ -1,22 +1,55 @@
 GO_FILES?=$$(find . -name '*.go' |grep -v vendor)
-TAG?=latest
 
 .GIT_COMMIT=$(shell git rev-parse HEAD)
-.GIT_VERSION=$(shell git describe --tags 2>/dev/null || echo "$(.GIT_COMMIT)")
-.GIT_UNTRACKEDCHANGES := $(shell git status --porcelain --untracked-files=no)
-ifneq ($(.GIT_UNTRACKEDCHANGES),)
-	.GIT_COMMIT := $(.GIT_COMMIT)-dirty
-endif
+.GIT_VERSION=$(shell git describe --tags --always --dirty)
+.LDFLAGS=-s -w -X main.Version=$(.GIT_VERSION) -X main.GitCommit=$(.GIT_COMMIT)
+.PLATFORMS=linux/amd64,linux/arm/v6,linux/arm64
+.BUILDX_OUTPUT=type=image,push=false
+
+# docker manifest command will work with Docker CLI 18.03 or newer
+# but for now it's still experimental feature so we need to enable that
+export DOCKER_CLI_EXPERIMENTAL=enabled
 
 export GOFLAGS=-mod=vendor
 
 .PHONY: build
 build:
-	./build.sh
+	@docker buildx create --use --name=multiarch --node multiarch
+	docker buildx build \
+		--progress=plain \
+		--build-arg http_proxy=${http_proxy} --build-arg https_proxy=${https_proxy} \
+		--build-arg VERSION=$(.GIT_VERSION) --build-arg GIT_COMMIT=$(.GIT_COMMIT) \
+		--platform $(.PLATFORMS) \
+		--output "$(.BUILDX_OUTPUT)" \
+		--target release \
+		--tag openfaas/faas-cli:latest \
+		--tag openfaas/faas-cli:$(.GIT_VERSION) .
+	docker buildx build \
+		--progress=plain \
+		--build-arg http_proxy=${http_proxy} --build-arg https_proxy=${https_proxy} \
+		--build-arg VERSION=$(.GIT_VERSION) --build-arg GIT_COMMIT=$(.GIT_COMMIT) \
+		--platform $(.PLATFORMS) \
+		--output "$(.BUILDX_OUTPUT)" \
+		--target root \
+		--tag openfaas/faas-cli:latest-root \
+		--tag openfaas/faas-cli:$(.GIT_VERSION)-root .
 
 .PHONY: build_redist
 build_redist:
```

Comment:
Previously we had a build that might have run 2x as fast because we built in Docker, then pulled out the binaries.

How that we are both building in Docker, and on the host, how does the performance compare?

Should we adapt the Docker builds to pull in these binaries peharps?

Reply:
I just copied the flow from inlets, i figured you preferred that

Reply:
What is the timing difference?

Reply:
i'll just revert. I don't know the timing difference because i haven't gotten a successful build. My goal was consistency between projects to reduce the maintenance effort. But I  don't have strong feelings here.

Reply:
I hav reverted that change

Reply:
Using buildx results in build time: 8 min 51 sec
The last master build was : 4 min 46 sec